src/python/zquantum/core/circuits/_serde.py

```diff
@@ -17,8 +18,21 @@ def serialize_expr(expr: sympy.Expr):
     return str(expr)
 
 
-def _make_symbols_map(symbol_names):
-    return {name: sympy.Symbol(name) for name in symbol_names}
+def _make_symbols_map(
+    symbol_names: Iterable[str],
+) -> Dict[str, Union[sympy.Symbol, Dict[int, sympy.Symbol]]]:
+    symbols_map: Dict[str, Union[sympy.Symbol, Dict[int, sympy.Symbol]]] = {}
+    for name in symbol_names:
+        # Check if the symbol name has brackets, such as "x[4]"
```

Comment:
**suggestion**
It would be nice to add a comment stating the reason we look for brackets, WDYT about sth like:
```suggestion
        # Check if the symbol name has brackets, such as "x[4]". This is a workaround
        # for importing circuits from Qiskit v0.28 when the circuit is parametrized with
        # symbols from a "ParameterVector".
```
?

Reply:
Added similar comment in f32a85b80bb929fe6087ca3e510c236d81484b59